ruby/all-but-the-first-element-from-array.md

```diff
@@ -0,0 +1,40 @@
+# Getting all but the first element from an Array
+
+Every now and then I need all items in an array except for the first one. I
+usually avoid `shift` because it mutates the array. I have been using `slice`
+but having to specify a specific indices or pass the length of the array is ugly
+and lower level than I'd like to go.
```

Comment:
Could these intro paragraphs be shortened or removed if you made the points in the comments before the invocations?

Example:

In Ruby, we have a few options to get all items in an array except for the first one.

Given this array:

I recently discovered `drop(n)` which will drop the first `n` elements from the
front of the array and return the rest.

```ruby
list = [1,2,3,4]
```

We could use [`shift`](http://www.ruby-doc.org/core-2.2.0/Array.html#method-i-shift), but it mutates the array:

```ruby
list.shift
=> 1
list
=> [2, 3, 4]
```

We could use [`slice`](http://www.ruby-doc.org/core-2.2.0/Array.html#method-i-slice), but specifying indices or passing the length of the array is ugly and low level:

```
list[1..-1]
# => [2,3,4]
list.slice(1, list.length)
# => [2,3,4]
```

My new favorite is [`drop`](http://www.ruby-doc.org/core-2.2.0/Array.html#method-i-drop), which does not mutate and does not require ugly, low-level code:

```
list.drop(1)
# => [2,3,4]
list
# => [1,2,3,4]
```

An added benefit of `drop` is that it will always return an array, while `slice` will sometimes return `nil`

```ruby
[].slice(1..-1)
# => nil
[].drop(1)
# => []
```